### PR TITLE
thcrap-steam-proton-wrapper: init at 0-unstable-2024-04-03

### PR DESCRIPTION
**Pull request by ashuramaruzxc**
## Description of changes
Add package with a [wrapper script](https://github.com/tactikauan/thcrap-steam-proton-wrapper) for running Touhou games under steam proton with: 
[thcrap patch](https://github.com/thpatch/thcrap) (English localization, QOL and etc),
[thprac](https://github.com/touhouworldcup/thprac) (Practice tool), 
vsync patch.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
```
1 package built:
thcrap-steam-proton-wrapper
```
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  bash,
+  subversion,
+  gnome,
+}:
+with lib; let
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:10]: Top-level `with` is discouraged, see [this draft PR](https://github.com/NixOS/nixpkgs/pull/294383) and [this guide](https://nix.dev/guides/best-practices#with-scopes) for some reasoning.

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "tactikauan";
+      repo = "thcrap-steam-proton-wrapper";
+      rev = version;
+      sha256 = "sha256-J2O8F75NMdsxSaNVr8zLf+vLEJE6CHqWQIIscuuJZ3o=";
+    };
+
+    buildInputs = [bash subversion];
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:24]:
```suggestion
    buildInputs = [ subversion ];
```

`bash` should be in the standard environment by default

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    nativeBuildInputs = [
+      makeWrapper
+    ];
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:28]:
```suggestion
    nativeBuildInputs = [ makeWrapper ];
```
nit

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    installPhase = ''
+      mkdir -p $out/bin
+      cp thcrap_proton $out/bin/thcrap_proton
+    '';
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:32]:
```suggestion
    installPhase = ''
      runHook preInstall
      
      mkdir -p $out/bin
      cp thcrap_proton $out/bin/thcrap_proton
      
      runHook postInstall
    '';
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  bash,
+  subversion,
+  gnome,
+}:
+with lib; let
+  version = "2b636c3";
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:11]:
```suggestion
  version = "0-unstable-2024-04-03";
```

See the [versioning conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming)

**Review by msanft — CHANGES_REQUESTED** [review]

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    mkdir -p $out/bin
+    cp thcrap_proton $out/bin/thcrap_proton
+
+    runHook postInstall
+  '';
+  postFixup = ''
+    wrapProgram $out/bin/thcrap_proton \
+      --prefix PATH : ${lib.makeBinPath [bash subversion gnome.zenity]}
+  '';
+  meta = with lib; {
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:36]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
  
  installPhase = ''
    runHook preInstall

    mkdir -p $out/bin
    cp thcrap_proton $out/bin/thcrap_proton

    runHook postInstall
  '';
  
  postFixup = ''
    wrapProgram $out/bin/thcrap_proton \
      --prefix PATH : ${lib.makeBinPath [bash subversion gnome.zenity]}
  '';
  
  meta = with lib; {
```
nit

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  bash,
+  subversion,
+  gnome,
+}:
+stdenv.mkDerivation {
+  pname = "thcrap-proton";
+  version = "0-unstable-2024-04-03";
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:12]: Thanks for updating, maybe you can also edit the PR title accordingly

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix (hunk 6 above)
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:36]:
```suggestion
  meta = {
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  postFixup = ''
+    wrapProgram $out/bin/thcrap_proton \
+      --prefix PATH : ${lib.makeBinPath [bash subversion gnome.zenity]}
+  '';
+  meta = with lib; {
+    description = ''A wrapper script for launching the official Touhou games on Steam with patches through Proton on GNU/Linux'';
+    longDescription = ''For further usage please visit https://github.com/tactikauan/thcrap-steam-proton-wrapper'';
+    homepage = "https://github.com/tactikauan/thcrap-steam-proton-wrapper";
+    changelog = "https://github.com/tactikauan/thcrap-steam-proton-wrapper/${version}";
+    license = licenses.unlicense;
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:41]:
```suggestion
    license = lib.licenses.unlicense;
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    maintainers = with maintainers; [ashuramaruzxc];
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:42]:
```suggestion
    maintainers = with lib.maintainers; [ashuramaruzxc];
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+    runHook postInstall
+  '';
+  postFixup = ''
+    wrapProgram $out/bin/thcrap_proton \
+      --prefix PATH : ${lib.makeBinPath [bash subversion gnome.zenity]}
+  '';
+  meta = with lib; {
+    description = ''A wrapper script for launching the official Touhou games on Steam with patches through Proton on GNU/Linux'';
+    longDescription = ''For further usage please visit https://github.com/tactikauan/thcrap-steam-proton-wrapper'';
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:38]:
```suggestion
```
Doesn't really provide more value than setting `homepage`.

**Review by msanft — CHANGES_REQUESTED** [review]

**Comment by msanft** [comment]
Result of `nixpkgs-review pr 301845` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>thcrap-steam-proton-wrapper</li>
  </ul>
</details>

**Comment by msanft** [comment]
Also change the PR title format to `package: init at version` please

**Comment by msanft** [comment]
Binary appears to work correctly to me

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  version = "0-unstable-2024-04-03";
+
+  src = fetchFromGitHub {
+    owner = "tactikauan";
+    repo = "thcrap-steam-proton-wrapper";
+    rev = "2b636c3f5f1ce1b9b41f731aa9397aa68d2ce66b";
+    sha256 = "sha256-J2O8F75NMdsxSaNVr8zLf+vLEJE6CHqWQIIscuuJZ3o=";
+  };
+
+  buildInputs = [subversion];
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:21]:
```suggestion
  buildInputs = [ subversion ];
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+    mkdir -p $out/bin
+    cp thcrap_proton $out/bin/thcrap_proton
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/thcrap_proton \
+      --prefix PATH : ${lib.makeBinPath [bash subversion gnome.zenity]}
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:36]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ bash subversion gnome.zenity ]}
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+  '';
+
+  meta = {
+    description = ''A wrapper script for launching the official Touhou games on Steam with patches through Proton on GNU/Linux'';
+    homepage = "https://github.com/tactikauan/thcrap-steam-proton-wrapper";
+    license = lib.licenses.unlicense;
+    maintainers = with lib.maintainers; [ashuramaruzxc];
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:43]:
```suggestion
    maintainers = with lib.maintainers; [ ashuramaruzxc ];
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+    platforms = ["x86_64-linux" "aarch64-linux"];
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:44]:
```suggestion
    platforms = [ "x86_64-linux" "aarch64-linux" ];
```

**Review by msanft — APPROVED** [review]
Only some nits, and adjust the PR title please

**Review by msanft — APPROVED** [review]

**Comment by DontEatOreo** [comment]
8e0c692ce697dc588effaef6e7b517c64b19529a and 93efa69ec58fd4332bbfa66871ef61c8d686d331...99e296c41b4a5819636bdd9c20b14224dcfbb3ff are breaking [Commit conventions](https://github.com/NixOS/nixpkgs/blob/bc5ee2b8f8c1b6a0f1e80a103ccd77febe85f8cc/CONTRIBUTING.md#commit-conventions)

For 8e0c692ce697dc588effaef6e7b517c64b19529a

> - When adding yourself as maintainer in the same pull request, make a separate commit with the message `maintainers: add <handle>`.

For 93efa69ec58fd4332bbfa66871ef61c8d686d331...99e296c41b4a5819636bdd9c20b14224dcfbb3ff

> - If you have commits `pkg-name: oh, forgot to insert whitespace`: squash commits in this case. Use `git rebase -i`.

The last 4 commits: 93efa69ec58fd4332bbfa66871ef61c8d686d331...99e296c41b4a5819636bdd9c20b14224dcfbb3ff can be squashed into one and be reworded to `thcrap-steam-proton-wrapper: apply suggestions from code review`

Also reword 93efa69ec58fd4332bbfa66871ef61c8d686d331 to `thcrap-steam-proton-wrapper: add long description`

**Comment by ashuramaruzxc** [comment]
Okay, rebased and squashed commits, and reworded the commit about the version of the package and adding myself as maintainer.
@DontEatOreo

**Review by msanft — APPROVED** [review]

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/thcrap_proton \
+      --prefix PATH : ${lib.makeBinPath [bash subversion gnome.zenity]}
+  '';
+
+  meta = {
+    description = ''A wrapper script for launching the official Touhou games on Steam with patches through Proton on GNU/Linux'';
```
Comment (msanft) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:40]:
```suggestion
    description = "A wrapper script for launching the official Touhou games on Steam with patches through Proton on GNU/Linux";
```
nit

**Review by msanft — APPROVED** [review]

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix [hunk 16]
```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  bash,
+  subversion,
+  gnome,
+}:
+stdenv.mkDerivation {
```
Comment (DontEatOreo) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:10]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, makeWrapper
, bash
, subversion
, gnome
}:

stdenv.mkDerivation {
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix (hunk 11 above)
Comment (DontEatOreo) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:21]:
```suggestion
  buildInputs = [ subversion ];
```

File: pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix (hunk 14 above)
Comment (DontEatOreo) [inline pkgs/by-name/th/thcrap-steam-proton-wrapper/package.nix:44]:
```suggestion
    maintainers = with lib.maintainers; [ ashuramaruzxc ];
    platforms = [ "x86_64-linux" "aarch64-linux" ];
```

**Review by 4evy — CHANGES_REQUESTED** [review]

**Comment by ashuramaruzxc** [comment]
> @ashuramaruzxc You'll want to reword `format: use nixfmt-rfc-style instead of alejandra` into `thcrap-steam-proton-wrapper: init at 0-unstable-2024-04-03`

yep sorry my bad, used commit message from the zsh history

**Comment by DontEatOreo** [comment]
Adding yourself to the maintainers file needs to be a separate commit

**Review by 4evy — APPROVED** [review]

**Review by 4evy — APPROVED** [review]

**Pull request merged**